repose-aggregator/commons/utilities/src/test/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapperTest.scala

```diff
@@ -93,6 +94,7 @@ class HttpServletRequestWrapperTest extends FunSpec with BeforeAndAfter with Mat
       wrappedRequest.getHeaderNames.asScala.toList should contain theSameElementsAs headerMap.keys.filterNot( _ == "foo")
     }
 
+    //todo: fix this test -- it should not be case sensitive
```

Comment:
Right, header names are case insensitive. The header is still there, but add header works by using the name passed as the key in the map. In other words, the casing of the header name passed into addHeader is used rather than the original casing. We have to store it with one casing or another; my changes used the casing provided to the addHeader method, your code uses the casing of the header that existed before calling addHeader.

In what way is case checking needed elsewhere? We never returned the wrapped name from the interface anyway. Or are you talking about internal to this wrapper class? Wrapping made things messy and harder to follow for me, so I proposed this alternative.